File: old/sources/net.sf.j2s.java.org.eclipse.osgi/src/org/eclipse/core/runtime/internal/adaptor/EclipseEnvironmentInfo.java
/*******************************************************************************
 * Copyright (c) 2003, 2005 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package org.eclipse.core.runtime.internal.adaptor;

import java.util.*;
import org.eclipse.core.runtime.adaptor.EclipseAdaptorMsg;
import org.eclipse.osgi.service.environment.Constants;
import org.eclipse.osgi.service.environment.EnvironmentInfo;
import org.eclipse.osgi.util.NLS;

/**
 * Internal class.
 */
public class EclipseEnvironmentInfo implements EnvironmentInfo {
	private static EclipseEnvironmentInfo singleton;
	private static String nl;
	private static String os;
	private static String ws;
	private static String arch;
	static String[] allArgs;
	static String[] frameworkArgs;
	static String[] appArgs;

	// While we recognize the SunOS operating system, we change
	// this internally to be Solaris.
	private static final String INTERNAL_OS_SUNOS = "SunOS"; //$NON-NLS-1$
	private static final String INTERNAL_OS_LINUX = "Linux"; //$NON-NLS-1$
	private static final String INTERNAL_OS_MACOSX = "Mac OS"; //$NON-NLS-1$
	private static final String INTERNAL_OS_AIX = "AIX"; //$NON-NLS-1$
	private static final String INTERNAL_OS_HPUX = "HP-UX"; //$NON-NLS-1$
	private static final String INTERNAL_OS_QNX = "QNX"; //$NON-NLS-1$

	// While we recognize the i386 architecture, we change
	// this internally to be x86.
	private static final String INTERNAL_ARCH_I386 = "i386"; //$NON-NLS-1$
	// While we recognize the amd64 architecture, we change
	// this internally to be x86_64.
	private static final String INTERNAL_AMD64 = "amd64"; //$NON-NLS-1$

	private EclipseEnvironmentInfo() {
		super();
		setupSystemContext();
	}

	public static EclipseEnvironmentInfo getDefault() {
		if (singleton == null)
			singleton = new EclipseEnvironmentInfo();
		return singleton;
	}

	public boolean inDevelopmentMode() {
		return System.getProperty("osgi.dev") != null; //$NON-NLS-1$
	}

	public boolean inDebugMode() {
		return System.getProperty("osgi.debug") != null; //$NON-NLS-1$
	}

	public String[] getCommandLineArgs() {
		return allArgs;
	}

	public String[] getFrameworkArgs() {
		return frameworkArgs;
	}

	public String[] getNonFrameworkArgs() {
		return appArgs;
	}

	public String getOSArch() {
		return arch;
	}

	public String getNL() {
		return nl;
	}

	public String getOS() {
		return os;
	}

	public String getWS() {
		return ws;
	}

	/**
	 * Initializes the execution context for this run of the platform.  The context
	 * includes information about the locale, operating system and window system.
	 * 
	 * NOTE: The OS, WS, and ARCH values should never be null. The executable should
	 * be setting these values and therefore this code path is obsolete for Eclipse
	 * when run from the executable.
	 */
	private static void setupSystemContext() {
		// if the user didn't set the locale with a command line argument then use the default.
		nl = System.getProperty("osgi.nl"); //$NON-NLS-1$
		if (nl != null) {
			StringTokenizer tokenizer = new StringTokenizer(nl, "_"); //$NON-NLS-1$
			int segments = tokenizer.countTokens();
			try {
				Locale userLocale = null;
				switch (segments) {
					case 1 :
						// use the 2 arg constructor to maintain compatibility with 1.3.1
						userLocale = new Locale(tokenizer.nextToken(), ""); //$NON-NLS-1$
						break;
					case 2 :
						userLocale = new Locale(tokenizer.nextToken(), tokenizer.nextToken());
						break;
					case 3 :
						userLocale = new Locale(tokenizer.nextToken(), tokenizer.nextToken(), tokenizer.nextToken());
						break;
					default :
						// if the user passed us in a bogus value then log a message and use the default
						System.err.println(NLS.bind(EclipseAdaptorMsg.error_badNL, nl)); 
						userLocale = Locale.getDefault();
						break;
				}
				Locale.setDefault(userLocale);
				System.getProperties().put("osgi.nl.user", nl); //$NON-NLS-1$
			} catch (NoSuchElementException e) {
				// fall through and use the default
			}
		}
		nl = Locale.getDefault().toString();
		System.getProperties().put("osgi.nl", nl); //$NON-NLS-1$

		// if the user didn't set the operating system with a command line 
		// argument then use the default.
		os = System.getProperty("osgi.os"); //$NON-NLS-1$
		if (os == null) {
			os = guessOS(System.getProperty("os.name"));//$NON-NLS-1$);
			System.getProperties().put("osgi.os", os); //$NON-NLS-1$
		}

		// if the user didn't set the window system with a command line 
		// argument then use the default.
		ws = System.getProperty("osgi.ws"); //$NON-NLS-1$
		if (ws == null) {
			ws = guessWS(os);
			System.getProperties().put("osgi.ws", ws); //$NON-NLS-1$
		}

		// if the user didn't set the system architecture with a command line 
		// argument then use the default.
		arch = System.getProperty("osgi.arch"); //$NON-NLS-1$
		if (arch == null) {
			String name = System.getProperty("os.arch");//$NON-NLS-1$
			// Map i386 architecture to x86
			if (name.equalsIgnoreCase(INTERNAL_ARCH_I386))
				arch = Constants.ARCH_X86;
			// Map amd64 architecture to x86_64
			else if (name.equalsIgnoreCase(INTERNAL_AMD64))
				arch = Constants.ARCH_X86_64;
			else
				arch = name;
			System.getProperties().put("osgi.arch", arch); //$NON-NLS-1$			
		}
	}

	public static void setAllArgs(String[] allArgs) {
		if (EclipseEnvironmentInfo.allArgs == null)
			EclipseEnvironmentInfo.allArgs = allArgs;
	}

	public static void setAppArgs(String[] appArgs) {
		if (EclipseEnvironmentInfo.appArgs == null)
			EclipseEnvironmentInfo.appArgs = appArgs;
	}

	public static void setFrameworkArgs(String[] frameworkArgs) {
		if (EclipseEnvironmentInfo.frameworkArgs == null)
			EclipseEnvironmentInfo.frameworkArgs = frameworkArgs;
	}

	public static String guessWS(String os) {
		// setup default values for known OSes if nothing was specified
		if (os.equals(Constants.OS_WIN32))
			return Constants.WS_WIN32;
		if (os.equals(Constants.OS_LINUX))
			return Constants.WS_MOTIF;
		if (os.equals(Constants.OS_MACOSX))
			return Constants.WS_CARBON;
		if (os.equals(Constants.OS_HPUX))
			return Constants.WS_MOTIF;
		if (os.equals(Constants.OS_AIX))
			return Constants.WS_MOTIF;
		if (os.equals(Constants.OS_SOLARIS))
			return Constants.WS_MOTIF;
		if (os.equals(Constants.OS_QNX))
			return Constants.WS_PHOTON;
		return Constants.WS_UNKNOWN;
	}

	public static String guessOS(String osName) {
		// check to see if the OS name is "Windows 98" or some other
		// flavour which should be converted to win32.
		if (osName.regionMatches(true, 0, Constants.OS_WIN32, 0, 3))
			return Constants.OS_WIN32;
		// EXCEPTION: All mappings of SunOS convert to Solaris
		if (osName.equalsIgnoreCase(INTERNAL_OS_SUNOS))
			return Constants.OS_SOLARIS;
		if (osName.equalsIgnoreCase(INTERNAL_OS_LINUX))
			return Constants.OS_LINUX;
		if (osName.equalsIgnoreCase(INTERNAL_OS_QNX))
			return Constants.OS_QNX;
		if (osName.equalsIgnoreCase(INTERNAL_OS_AIX))
			return Constants.OS_AIX;
		if (osName.equalsIgnoreCase(INTERNAL_OS_HPUX))
			return Constants.OS_HPUX;
		// os.name on Mac OS can be either Mac OS or Mac OS X
		if (osName.regionMatches(true, 0, INTERNAL_OS_MACOSX, 0, INTERNAL_OS_MACOSX.length()))
			return Constants.OS_MACOSX;
		return Constants.OS_UNKNOWN;
	}
}

File: old/incubator/net.sf.j2s.test.mauve/j2slib/org/eclipse/core/runtime/internal/adaptor/EclipseEnvironmentInfo.js
Clazz.declarePackage ("org.eclipse.core.runtime.internal.adaptor");
Clazz.load (["org.eclipse.osgi.service.environment.EnvironmentInfo"], "org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo", ["java.util.Locale", "$.StringTokenizer", "org.eclipse.core.runtime.adaptor.EclipseAdaptorMsg", "org.eclipse.osgi.util.NLS"], function () {
c$ = Clazz.declareType (org.eclipse.core.runtime.internal.adaptor, "EclipseEnvironmentInfo", null, org.eclipse.osgi.service.environment.EnvironmentInfo);
Clazz.makeConstructor (c$, 
($fz = function () {
org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.setupSystemContext ();
}, $fz.isPrivate = true, $fz));
c$.getDefault = Clazz.defineMethod (c$, "getDefault", 
function () {
if (org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.singleton == null) ($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.singleton =  new org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo (), org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.singleton = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.singleton, $t$);
return org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.singleton;
});
Clazz.overrideMethod (c$, "inDevelopmentMode", 
function () {
return System.getProperty ("osgi.dev") != null;
});
Clazz.overrideMethod (c$, "inDebugMode", 
function () {
return System.getProperty ("osgi.debug") != null;
});
Clazz.overrideMethod (c$, "getCommandLineArgs", 
function () {
return org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.allArgs;
});
Clazz.overrideMethod (c$, "getFrameworkArgs", 
function () {
return org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.frameworkArgs;
});
Clazz.overrideMethod (c$, "getNonFrameworkArgs", 
function () {
return org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.appArgs;
});
Clazz.overrideMethod (c$, "getOSArch", 
function () {
return org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.arch;
});
Clazz.overrideMethod (c$, "getNL", 
function () {
return org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.nl;
});
Clazz.overrideMethod (c$, "getOS", 
function () {
return org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.os;
});
Clazz.overrideMethod (c$, "getWS", 
function () {
return org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.ws;
});
c$.setupSystemContext = Clazz.defineMethod (c$, "setupSystemContext", 
($fz = function () {
($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.nl = System.getProperty ("osgi.nl"), org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.nl = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.nl, $t$);
if (org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.nl != null) {
var tokenizer =  new java.util.StringTokenizer (org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.nl, "_");
var segments = tokenizer.countTokens ();
try {
var userLocale = null;
switch (segments) {
case 1:
userLocale =  new java.util.Locale (tokenizer.nextToken (), "");
break;
case 2:
userLocale =  new java.util.Locale (tokenizer.nextToken (), tokenizer.nextToken ());
break;
case 3:
userLocale =  new java.util.Locale (tokenizer.nextToken (), tokenizer.nextToken (), tokenizer.nextToken ());
break;
default:
System.err.println (org.eclipse.osgi.util.NLS.bind (org.eclipse.core.runtime.adaptor.EclipseAdaptorMsg.error_badNL, org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.nl));
userLocale = java.util.Locale.getDefault ();
break;
}
java.util.Locale.setDefault (userLocale);
System.getProperties ().put ("osgi.nl.user", org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.nl);
} catch (e) {
if (Clazz.instanceOf (e, java.util.NoSuchElementException)) {
} else {
throw e;
}
}
}($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.nl = java.util.Locale.getDefault ().toString (), org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.nl = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.nl, $t$);
System.getProperties ().put ("osgi.nl", org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.nl);
($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.os = System.getProperty ("osgi.os"), org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.os = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.os, $t$);
if (org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.os == null) {
($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.os = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.guessOS (System.getProperty ("os.name")), org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.os = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.os, $t$);
System.getProperties ().put ("osgi.os", org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.os);
}($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.ws = System.getProperty ("osgi.ws"), org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.ws = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.ws, $t$);
if (org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.ws == null) {
($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.ws = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.guessWS (org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.os), org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.ws = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.ws, $t$);
System.getProperties ().put ("osgi.ws", org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.ws);
}($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.arch = System.getProperty ("osgi.arch"), org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.arch = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.arch, $t$);
if (org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.arch == null) {
var name = System.getProperty ("os.arch");
if (name.equalsIgnoreCase ("i386")) ($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.arch = "x86", org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.arch = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.arch, $t$);
 else if (name.equalsIgnoreCase ("amd64")) ($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.arch = "x86_64", org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.arch = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.arch, $t$);
 else ($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.arch = name, org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.arch = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.arch, $t$);
System.getProperties ().put ("osgi.arch", org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.arch);
}}, $fz.isPrivate = true, $fz));
c$.setAllArgs = Clazz.defineMethod (c$, "setAllArgs", 
function (allArgs) {
if (org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.allArgs == null) ($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.allArgs = allArgs, org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.allArgs = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.allArgs, $t$);
}, "~A");
c$.setAppArgs = Clazz.defineMethod (c$, "setAppArgs", 
function (appArgs) {
if (org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.appArgs == null) ($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.appArgs = appArgs, org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.appArgs = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.appArgs, $t$);
}, "~A");
c$.setFrameworkArgs = Clazz.defineMethod (c$, "setFrameworkArgs", 
function (frameworkArgs) {
if (org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.frameworkArgs == null) ($t$ = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.frameworkArgs = frameworkArgs, org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.prototype.frameworkArgs = org.eclipse.core.runtime.internal.adaptor.EclipseEnvironmentInfo.frameworkArgs, $t$);
}, "~A");
c$.guessWS = Clazz.defineMethod (c$, "guessWS", 
function (os) {
if (os.equals ("win32")) return "win32";
if (os.equals ("linux")) return "motif";
if (os.equals ("macosx")) return "carbon";
if (os.equals ("hpux")) return "motif";
if (os.equals ("aix")) return "motif";
if (os.equals ("solaris")) return "motif";
if (os.equals ("qnx")) return "photon";
return "unknown";
}, "~S");
c$.guessOS = Clazz.defineMethod (c$, "guessOS", 
function (osName) {
if (osName.regionMatches (true, 0, "win32", 0, 3)) return "win32";
if (osName.equalsIgnoreCase ("SunOS")) return "solaris";
if (osName.equalsIgnoreCase ("Linux")) return "linux";
if (osName.equalsIgnoreCase ("QNX")) return "qnx";
if (osName.equalsIgnoreCase ("AIX")) return "aix";
if (osName.equalsIgnoreCase ("HP-UX")) return "hpux";
if (osName.regionMatches (true, 0, "Mac OS", 0, "Mac OS".length)) return "macosx";
return "unknown";
}, "~S");
Clazz.defineStatics (c$,
"singleton", null,
"nl", null,
"os", null,
"ws", null,
"arch", null,
"allArgs", null,
"frameworkArgs", null,
"appArgs", null,
"INTERNAL_OS_SUNOS", "SunOS",
"INTERNAL_OS_LINUX", "Linux",
"INTERNAL_OS_MACOSX", "Mac OS",
"INTERNAL_OS_AIX", "AIX",
"INTERNAL_OS_HPUX", "HP-UX",
"INTERNAL_OS_QNX", "QNX",
"INTERNAL_ARCH_I386", "i386",
"INTERNAL_AMD64", "amd64");
});
